src/Namotion.Reflection/TypeExtensions.cs

```diff
@@ -75,10 +75,11 @@ public static bool InheritsFromTypeName(this Type type, string typeName, TypeNam
         /// <summary>Gets the type of the array item.</summary>
         public static Type? GetEnumerableItemType(this Type type)
         {
-            var elementType = type.GetElementType();
-            if (elementType != null)
+            var genericTypeArguments = type.GetGenericTypeArgumentsOfTypeOrBaseTypes();
```

Comment:
Why this change, and why is it different from the EnumerableItemType property now ?

Reply:
We use generic type arguments also for EnumerableItemType now, so that your tests work... that's why I'd also keep it here... also because it's probably faster and the same as before, no?

Reply:
In theory we should probably check in both versions whether there is a GetEnumerator method?

Reply:
Don't understand your last question. Do you want to chat on gitter?